lib/libc/minimal/Kconfig

```diff
@@ -14,31 +14,43 @@ config MINIMAL_LIBC_NON_REENTRANT_FUNCTIONS
 	  option may require an additional memory protection region.
 
 config MINIMAL_LIBC_MALLOC
-	bool "Minimal libc malloc implementation"
+	bool "[DEPRECATED] Minimal libc malloc implementation"
 	default y
+	select COMMON_LIBC_MALLOC
 	help
+	  [DEPRECATED] Use COMMON_LIBC_MALLOC
```

Comment:
Deprecated Kconfigs should `select DEPRECATED`.

Reply:
I don't know how to make that work. Because this configuration entry (along with the finer grained options below this) is `default y`, applications would probably only ever explicitly set this to `n`. We want to catch those applications and warn them that their configuration selection has been deprecated, as well as warning applications which might explicitly set this to `y` while not warning applications with no setting at all.

Reply:
Fair enough.